torch/onnx/symbolic_opset9.py

```diff
@@ -6429,10 +6387,9 @@ def prim_unchecked_cast(g: jit_utils.GraphContext, self):
 @_onnx_symbolic("prim::dtype")
 @_beartype.beartype
 def prim_dtype(g: jit_utils.GraphContext, self):
-    scalar_name = symbolic_helper._try_get_scalar_type(self)
-    if scalar_name is None:
-        scalar_name = "Float"
-    scalar_type = _type_utils.JitScalarType.from_name(scalar_name)
+    scalar_type = symbolic_helper._try_get_scalar_type(self)
```

Comment:
non blocking: when _try_get_scalar_type takes one argument, I think it can be done cleanly with `from_value`:

```python
scalar_type = _type_utils.JitScalarType.from_value(self, default=_type_utils.JitScalarType.FLOAT)
```

This can be applied to other places where _try_get_scalar_type is used on a single argument

Reply:
yes, there is more than one way of doing it.
The current code fixes all bugs scalar type bugs and as a bonus did a lot of refactoring that the original `JitScalarType` didn't do for some reason.